modules/util/source-formatter/src/main/java/com/liferay/source/formatter/check/UpgradeJavaDDMFormValuesSerializerTrackerCheck.java

```diff
@@ -0,0 +1,113 @@
+/**
+ * SPDX-FileCopyrightText: (c) 2023 Liferay, Inc. https://liferay.com
+ * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
+ */
+
+package com.liferay.source.formatter.check;
+
+import com.liferay.petra.string.CharPool;
+import com.liferay.petra.string.StringBundler;
+import com.liferay.portal.kernel.util.StringUtil;
+import com.liferay.source.formatter.parser.JavaClass;
+import com.liferay.source.formatter.parser.JavaClassParser;
+import com.liferay.source.formatter.parser.JavaMethod;
+import com.liferay.source.formatter.parser.JavaTerm;
+
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * @author Nícolas Moura
+ */
+public class UpgradeJavaDDMFormValuesSerializerTrackerCheck
+	extends BaseUpgradeCheck {
+
+	@Override
+	protected String format(
+			String fileName, String absolutePath, String content)
+		throws Exception {
+
+		if (!content.contains("DDMFormValuesSerializerTracker")) {
+			return content;
+		}
+
+		Matcher trackerReferenceMatcher = _trackerReferencePattern.matcher(
+			content);
+
+		String newContent = content;
+
+		if (trackerReferenceMatcher.find()) {
+			newContent = StringUtil.removeSubstring(
+				content, trackerReferenceMatcher.group());
+
+			newContent = StringUtil.removeSubstring(
+				newContent,
+				"import com.liferay.dynamic.data.mapping.io." +
+					"DDMFormValuesSerializerTracker;\n");
+		}
+
+		JavaClass javaClass = JavaClassParser.parseJavaClass(
+			fileName, newContent);
+
+		for (JavaTerm childJavaTerm : javaClass.getChildJavaTerms()) {
+			if (!childJavaTerm.isJavaMethod()) {
+				continue;
+			}
+
+			JavaMethod javaMethod = (JavaMethod)childJavaTerm;
+
+			String javaMethodContent = javaMethod.getContent();
+
+			if (javaMethodContent.contains(
+					"setDDMFormValuesSerializerTracker")) {
+
+				// Checar se vem a anotação no javaMethodContent
```

Comment:
@nicolas-sss, please remove this comment

Reply:
Sorry! I will do it